src/parser/mod.rs

```diff
@@ -10589,9 +10580,23 @@ impl<'a> Parser<'a> {
             name,
             data_type,
             options,
+            options_comma_separated: !dialect_of!(self is SnowflakeDialect),
```

Comment:
Instead of introducing a new flag, can we change the datatype of `options` to be an enum that encodes the same information? Something like the following
```rust
enum ColumnOptions {
    CommaSeparated(Vec<ColumnOption>),
    SpaceSeparated(Vec<ColumnOption>)
}

options: Option<ColumnOptions>
```